gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -750,9 +751,14 @@ else if (desiredTargetExistingPaths.size() > 0) {
 
   private void checkPartitionedTableCompatibility(Table desiredTargetTable, Table existingTargetTable)
       throws IOException {
-    if (!desiredTargetTable.getDataLocation().equals(existingTargetTable.getDataLocation())) {
-      throw new HiveTableLocationNotMatchException(desiredTargetTable.getDataLocation(),
-          existingTargetTable.getDataLocation());
+    try {
+      if (!this.targetFs.resolvePath(desiredTargetTable.getDataLocation())
+          .equals(this.targetFs.resolvePath(existingTargetTable.getDataLocation()))) {
+        throw new HiveTableLocationNotMatchException(desiredTargetTable.getDataLocation(), existingTargetTable.getDataLocation());
```

Comment:
the exception args no longer reflect the equality operands.  could this message become confusing (e.g. if the two do match on their own, but however didn't upon mapping through `this.targetFs.resolvePath()`)?

Reply:
I think a scenario where the paths match exactly but fail when using `resolvePath()` should never happen, since resolvePath in this scenario would just be validating their FS is equivalent (logical vs physical paths). If it fails the `resolvePath()`, there should still be some noticeable difference in the paths such that the user can discern some sort of meaning.